orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy

```diff
@@ -599,7 +599,7 @@ class JedisExecutionRepositorySpec extends ExecutionRepositoryTck<JedisExecution
     }
 
     when:
-    def previousRepository = new JedisExecutionRepository(new NoopRegistry(), previousRedisClientDelegate.get(), Optional.empty(), 1, 50)
+    def previousRepository = new JedisExecutionRepository(new RedisClientSelector([redisClientDelegate, previousRedisClientDelegate]), 1, 50)
     previousRepository.store(orchestration)
```

Comment:
Not clear to me whether this actually writes to the previous redis. Why wouldn't `this.redisClientDelegate = redisClientSelector.primary(EXECUTION_REPOSITORY);` for `previousRepository` select `redisClientDelegate` as it's named primaryDefault? It looks like the test will incorrectly pass if previousRepository actually points to the current. Adding a repository object that only has the primary wired and asserting that it doesn't contain the orchestration would verify correctness.

Reply:
You're right. I buggered this test.